app/components/views/PKlima/index.tsx

```diff
@@ -104,20 +129,28 @@ export const PKlima: FC<Props> = (props) => {
     try {
       const value = quantity.toString();
       setQuantity("");
-      await exerciseTransaction({
+      const approvedValue = await exerciseTransaction({
         value,
         provider,
         onStatus: setStatus,
       });
-      dispatch(redeemPklima(value));
+      dispatch(redeemPklima(approvedValue));
+      dispatch(
+        decrementAllowance({
+          token: "bct",
+          spender: "pklima_exercise",
+          value: approvedValue,
```

Comment:
But this is only true for future users.
All others which have once allowed the 1.000.000 gwei has to decrement their value each time.
Otherwise, if those users would immediately do another transaction, do not have an approval no more. Yes/No?

Reply:
True, the way the UIs are built now, it doesnt do the approval flow if they already have an approval. And I guess the new transaction modal would check for an existing approval and skip that step if they did it already.

But I'm thinking now that it would be better to move all users to the new "as-needed" approvals. So when we build our new flow, it's a simple check:
```if (allowance !== value) { showApprovalStep() }```

Reply:
I updated this issue to clarify that, also added a point about multiple-approvals
https://github.com/KlimaDAO/klimadao/issues/9